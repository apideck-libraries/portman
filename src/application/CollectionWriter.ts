import { CollectionDefinition } from 'postman-collection'
import {
  injectEnvVariables,
  orderCollectionRequests,
  overwriteCollectionKeyValues,
  overwriteCollectionValues,
  overwriteCollectionSecurityValues,
  writeCollectionPreRequestScripts,
  writeRawReplacements
} from '.'
import { GlobalConfig, PortmanConfig, PortmanOptions } from '../types'
import { isEmptyObject } from 'utils'

export class CollectionWriter {
  public collection: CollectionDefinition
  config: PortmanConfig
  options: PortmanOptions

  constructor(config: PortmanConfig, options: PortmanOptions, collection: CollectionDefinition) {
    this.config = config
    this.options = options
    this.collection = collection
  }

  public execute(): void {
    const { envFile, baseUrl } = this.options

    const { globals = {} } = this.config
    const {
      collectionPreRequestScripts = [],
      securityOverwrites = {},
      keyValueReplacements = {},
      valueReplacements = {},
      rawReplacements = [],
      orderOfOperations = []
    } = globals as GlobalConfig

    let collection = this.collection

    // --- Portman - Set Security values for Postman
    if (securityOverwrites && !isEmptyObject(securityOverwrites)) {
      collection = overwriteCollectionSecurityValues(collection, securityOverwrites)
    }

    // --- Portman - Search for keys in dictionary to set the value if key is found anywhere in collection
    if (keyValueReplacements && !isEmptyObject(keyValueReplacements)) {
      collection = overwriteCollectionKeyValues(collection, keyValueReplacements)
    }

    // --- Portman - Search for keys in dictionary to set the values if value is found anywhere in collection
    if (valueReplacements && !isEmptyObject(valueReplacements)) {
      collection = overwriteCollectionValues(collection, valueReplacements)
    }

    // --- Portman - Use .env to inject variables on Collection
    if (envFile) {
      collection = injectEnvVariables(collection, envFile, baseUrl)
    }

    // --- Portman - Set manually order Postman requests
    if (orderOfOperations && orderOfOperations.length > 0) {
      collection = orderCollectionRequests(collection, orderOfOperations)
    }

    // --- Portman - Set Postman pre-requests
    if (collectionPreRequestScripts && collectionPreRequestScripts.length > 0) {
      collection = writeCollectionPreRequestScripts(collection, collectionPreRequestScripts)
    }

    // --- Portman - Replace & clean-up Postman
    if (rawReplacements && rawReplacements.length > 0) {
      const collectionString = writeRawReplacements(
        JSON.stringify(collection, null, 2),
        rawReplacements
      )
      this.collection = JSON.parse(collectionString)
    } else {
      this.collection = collection
    }
  }
}
