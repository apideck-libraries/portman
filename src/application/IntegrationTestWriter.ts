import { Collection, Item, ItemGroup } from 'postman-collection'
import { IntegrationTestConfig, VariationConfig, PortmanTestTypes } from '../types'
import { TestSuite, VariationWriter } from './'

export class IntegrationTestWriter {
  public testSuite: TestSuite
  integrationTestFolder: ItemGroup<Item>
  integrationTestCollection: Collection

  constructor({
    testSuite,
    integrationTestFolderName
  }: {
    testSuite: TestSuite
    integrationTestFolderName: string
  }) {
    this.testSuite = testSuite
    this.integrationTestFolder = new ItemGroup<Item>({
      name: integrationTestFolderName
    })
    this.integrationTestCollection = new Collection()
  }

  public add(integrationTest: IntegrationTestConfig): void {
    const { testSuite } = this
    const { name, operations } = integrationTest

    const variationWriter = new VariationWriter({
      testSuite: this.testSuite,
      variationFolderName: name
    })

    operations.map(operation => {
      const variations = (operation?.variations || []) as VariationConfig[]
      const { openApiOperationId } = operation

      const pmOperation = testSuite.postmanParser.getOperationById(openApiOperationId)

      if (!pmOperation) return

      const oaOperation = testSuite.oasParser.getOperationByPath(pmOperation.pathRef)

      // Handle integration variations
      variations.map(variation => {
        const variationName = variation.name

        const operationVariation = pmOperation.clone({
          newId: `${variationName}-${Math.random()}`,
          name: variationName
        })

        // Set/Update Portman operation test type
        this.testSuite.registerOperationTestType(operationVariation, PortmanTestTypes.integration)

        variationWriter.injectVariations(operationVariation, oaOperation, variation, operation)
        variationWriter.addToFolder(operationVariation, variationWriter.variationFolder)
      })
    })

    this.integrationTestCollection = variationWriter.mergeToCollection(
      this.integrationTestCollection
    )
  }

  public mergeToCollection(collection: Collection): Collection {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    this.integrationTestCollection.items.map((item: any) => {
      this.integrationTestFolder.items.add(item)
    })

    collection.items.add(this.integrationTestFolder)
    return collection
  }
}
