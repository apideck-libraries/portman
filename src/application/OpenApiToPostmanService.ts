import fs from 'fs-extra'
import oaConverter from 'openapi-to-postmanv2'
import ora from 'ora'
import path from 'path'

export class OpenApiToPostmanService {
  /**
   * Helper function for the CLI to convert OpenApi data input
   * (ported from https://github.com/postmanlabs/openapi-to-postman)
   * @param {String} openApiObj - OpenApi spec file used for conversion input
   * @param options - OpenApi to Postman conversion options
   * @returns {string}
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
  async convert(openApiObj: string, options: any): Promise<string> {
    return await new Promise((resolve, reject) => {
      const inputFile = path.resolve(options.inputFile)
      openApiObj = fs.readFileSync(inputFile, 'utf8')

      // apply options from config file if present
      if (options.configFile) {
        const configFile = path.resolve(options.configFile)
        // console.log('Options Config file: ', configFile)
        const configFileOptions = JSON.parse(fs.readFileSync(configFile, 'utf8'))
        options = Object.assign({}, options, configFileOptions)
      }

      if (options.testsuiteFile) {
        const testsuiteFile = path.resolve(options.testsuiteFile)
        // console.log('Testsuite file: ', testsuiteFile)
        options.testSuite = true
        options.testSuiteSettings = JSON.parse(fs.readFileSync(testsuiteFile, 'utf8'))
      }

      const spinner = ora({
        prefixText: ' ',
        text: 'Converting OpenApi to Postman Collection'
      }).start()

      // Convert OpenApi to Postman collection
      oaConverter.convert(
        {
          type: 'string',
          data: openApiObj
        },
        options,
        (err, status) => {
          if (err) {
            spinner.fail(err.toString())
            reject(err)
          }
          if (!status.result) {
            // console.log(status.reason)
            spinner.fail(status.reason)
            reject(status.reason)
          } else if (options.outputFile) {
            const filePath = path.resolve(options.outputFile)

            try {
              fs.writeFileSync(filePath, JSON.stringify(status.output[0].data, null, 4))
            } catch (error) {
              console.log('Could not write to file', error)
              spinner.fail(error.toString())
              reject(error)
            }

            // Return Postman collection
            spinner.succeed('Conversion successful')
            resolve(status.output[0].data)
          } else {
            spinner.succeed('Conversion successful')
            resolve(status.output[0].data)
          }
        }
      )
    })
  }
}
