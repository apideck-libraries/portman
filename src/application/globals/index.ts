export * from './injectEnvVariables'
export * from './orderCollectionRequests'
export * from './overwriteCollectionKeyValues'
export * from './overwriteCollectionSecurityValues'
export * from './overwriteCollectionValues'
export * from './renamePostmanCollection'
export * from './stripResponseExamples'
export * from './writeCollectionPreRequestScripts'
export * from './writeRawReplacements'
