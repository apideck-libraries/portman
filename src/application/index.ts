export * from './CollectionWriter'
export * from './globals'
export * from './newman'
export * from './overwrites'
export * from './tests'
export * from './TestSuite'
export * from './variables'
