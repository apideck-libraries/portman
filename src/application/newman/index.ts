export * from './runNewmanWith'
export * from './writeNewmanEnv'
