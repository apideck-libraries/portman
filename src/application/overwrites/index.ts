export * from './applyOverwrites'
export * from './overwriteRequestBody'
export * from './overwriteRequestHeaders'
export * from './overwriteRequestPathIdVariables'
export * from './overwriteRequestPathVariables'
export * from './overwriteRequestQueryParams'
export * from './overwriteRequestSecurity'
