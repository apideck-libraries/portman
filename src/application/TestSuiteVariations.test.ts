import * as Either from 'fp-ts/lib/Either'
import fs from 'fs-extra'
import { Collection } from 'postman-collection'
import { TestSuite, VariationWriter } from '../application'
import { getConfig } from '../lib'
import { OpenApiParser } from '../oas'
import { PostmanParser } from '../postman'
import { omitKeys, PortmanError } from '../utils'

describe('TestSuite Variations', () => {
  let postmanParser: PostmanParser
  let oasParser: OpenApiParser
  let testSuite: TestSuite

  const postmanJson = '__tests__/fixtures/crm.postman.json'
  const oasYml = '__tests__/fixtures/crm.yml'
  const portmanConfigFile = '__tests__/fixtures/portman-variations.crm.json'

  beforeEach(async () => {
    oasParser = new OpenApiParser()
    await oasParser.convert({ inputFile: oasYml })
    const postmanObj = JSON.parse(fs.readFileSync(postmanJson).toString())
    const configResult = await getConfig(portmanConfigFile)

    if (Either.isLeft(configResult)) {
      return PortmanError.render(configResult.left)
    }

    const config = configResult.right

    postmanParser = new PostmanParser({
      collection: new Collection(postmanObj),
      oasParser: oasParser
    })

    testSuite = new TestSuite({ oasParser, postmanParser, config })
    testSuite.variationWriter = new VariationWriter({
      testSuite: testSuite,
      variationFolderName: 'Variation Tests'
    })
  })

  it('should generateVariationTests for variations', () => {
    testSuite.generateVariationTests()
    expect(omitKeys(testSuite.collection.toJSON(), ['id', '_postman_id'])).toMatchSnapshot()
  })
})
