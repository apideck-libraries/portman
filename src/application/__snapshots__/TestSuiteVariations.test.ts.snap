// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite Variations openApiRequest content-types should handle wildcard request content-type and generate multiple variations 1`] = `
Object {
  "_": Object {
    "postman_id": "11111111-1111-1111-1111-111111111111",
  },
  "event": Array [],
  "info": Object {
    "name": "Request Multi",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "Create multi",
      "request": Object {
        "body": Object {
          "mode": "raw",
          "options": Object {
            "raw": Object {
              "language": "json",
            },
          },
          "raw": "{\\"value\\":\\"example\\"}",
        },
        "method": "POST",
        "url": Object {
          "host": Array [
            "{{baseUrl}}",
          ],
          "path": Array [
            "multi",
          ],
          "query": Array [],
          "variable": Array [],
        },
      },
      "response": Array [],
    },
    Object {
      "event": Array [],
      "item": Array [
        Object {
          "event": Array [],
          "item": Array [
            Object {
              "event": Array [
                Object {
                  "listen": "test",
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[POST]::/multi - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                    "type": "text/javascript",
                  },
                },
              ],
              "name": "Create multi[WildcardRequest][application/json]",
              "request": Object {
                "body": Object {
                  "mode": "raw",
                  "options": Object {
                    "raw": Object {
                      "language": "json",
                    },
                  },
                  "raw": "{\\"value\\":\\"example\\"}",
                },
                "header": Array [
                  Object {
                    "key": "Content-Type",
                    "value": "application/json",
                  },
                ],
                "method": "POST",
                "name": "Create multi[WildcardRequest][application/json]",
                "url": Object {
                  "host": Array [
                    "{{baseUrl}}",
                  ],
                  "path": Array [
                    "multi",
                  ],
                  "query": Array [],
                  "variable": Array [],
                },
              },
              "response": Array [],
            },
            Object {
              "event": Array [
                Object {
                  "listen": "test",
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[POST]::/multi - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                    "type": "text/javascript",
                  },
                },
              ],
              "name": "Create multi[WildcardRequest][text/plain]",
              "request": Object {
                "body": Object {
                  "mode": "raw",
                  "options": Object {
                    "raw": Object {
                      "language": "json",
                    },
                  },
                  "raw": "{\\"value\\":\\"example\\"}",
                },
                "header": Array [
                  Object {
                    "key": "Content-Type",
                    "value": "text/plain",
                  },
                ],
                "method": "POST",
                "name": "Create multi[WildcardRequest][text/plain]",
                "url": Object {
                  "host": Array [
                    "{{baseUrl}}",
                  ],
                  "path": Array [
                    "multi",
                  ],
                  "query": Array [],
                  "variable": Array [],
                },
              },
              "response": Array [],
            },
          ],
          "name": "Request Multi Variations",
        },
      ],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "type": "any",
      "value": "https://example.com",
    },
  ],
}
`;

exports[`TestSuite Variations openApiRequest content-types should use a specific request content-type when multiple are available 1`] = `
Object {
  "_": Object {
    "postman_id": "11111111-1111-1111-1111-111111111111",
  },
  "event": Array [],
  "info": Object {
    "name": "Request Multi",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "Create multi",
      "request": Object {
        "body": Object {
          "mode": "raw",
          "options": Object {
            "raw": Object {
              "language": "json",
            },
          },
          "raw": "{\\"value\\":\\"example\\"}",
        },
        "method": "POST",
        "url": Object {
          "host": Array [
            "{{baseUrl}}",
          ],
          "path": Array [
            "multi",
          ],
          "query": Array [],
          "variable": Array [],
        },
      },
      "response": Array [],
    },
    Object {
      "event": Array [],
      "item": Array [
        Object {
          "event": Array [],
          "item": Array [
            Object {
              "event": Array [
                Object {
                  "listen": "test",
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[POST]::/multi - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                    "type": "text/javascript",
                  },
                },
              ],
              "name": "Create multi[SpecificRequest]",
              "request": Object {
                "body": Object {
                  "mode": "raw",
                  "options": Object {
                    "raw": Object {
                      "language": "json",
                    },
                  },
                  "raw": "{\\"value\\":\\"example\\"}",
                },
                "header": Array [
                  Object {
                    "key": "Content-Type",
                    "value": "text/plain",
                  },
                ],
                "method": "POST",
                "name": "Create multi[SpecificRequest]",
                "url": Object {
                  "host": Array [
                    "{{baseUrl}}",
                  ],
                  "path": Array [
                    "multi",
                  ],
                  "query": Array [],
                  "variable": Array [],
                },
              },
              "response": Array [],
            },
          ],
          "name": "Request Multi Variations",
        },
      ],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "type": "any",
      "value": "https://example.com",
    },
  ],
}
`;

exports[`TestSuite Variations openApiResponse content-types should handle wildcard content-type and generate multiple variations 1`] = `
Object {
  "_": Object {
    "postman_id": "cccccccc-cccc-cccc-cccc-cccccccccccc",
  },
  "event": Array [],
  "info": Object {
    "name": "Minimal CRM",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "List companies",
      "request": Object {
        "header": Array [
          Object {
            "key": "x-apideck-consumer-id",
            "value": "test_consumer",
          },
          Object {
            "key": "x-apideck-app-id",
            "value": "2222",
          },
          Object {
            "key": "x-apideck-service-id",
            "value": "salesforce",
          },
        ],
        "method": "GET",
        "url": Object {
          "host": Array [
            "{{baseUrl}}",
          ],
          "path": Array [
            "crm",
            "companies",
          ],
          "query": Array [],
          "variable": Array [],
        },
      },
      "response": Array [],
    },
    Object {
      "event": Array [],
      "item": Array [
        Object {
          "event": Array [],
          "item": Array [
            Object {
              "event": Array [
                Object {
                  "listen": "test",
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                    "type": "text/javascript",
                  },
                },
              ],
              "name": "List companies[WildcardContentType][application/json]",
              "request": Object {
                "header": Array [
                  Object {
                    "key": "x-apideck-consumer-id",
                    "value": "test_consumer",
                  },
                  Object {
                    "key": "x-apideck-app-id",
                    "value": "2222",
                  },
                  Object {
                    "key": "x-apideck-service-id",
                    "value": "salesforce",
                  },
                  Object {
                    "key": "Accept",
                    "value": "application/json",
                  },
                ],
                "method": "GET",
                "name": "List companies[WildcardContentType][application/json]",
                "url": Object {
                  "host": Array [
                    "{{baseUrl}}",
                  ],
                  "path": Array [
                    "crm",
                    "companies",
                  ],
                  "query": Array [],
                  "variable": Array [],
                },
              },
              "response": Array [],
            },
            Object {
              "event": Array [
                Object {
                  "listen": "test",
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                    "type": "text/javascript",
                  },
                },
              ],
              "name": "List companies[WildcardContentType][text/plain]",
              "request": Object {
                "header": Array [
                  Object {
                    "key": "x-apideck-consumer-id",
                    "value": "test_consumer",
                  },
                  Object {
                    "key": "x-apideck-app-id",
                    "value": "2222",
                  },
                  Object {
                    "key": "x-apideck-service-id",
                    "value": "salesforce",
                  },
                  Object {
                    "key": "Accept",
                    "value": "text/plain",
                  },
                ],
                "method": "GET",
                "name": "List companies[WildcardContentType][text/plain]",
                "url": Object {
                  "host": Array [
                    "{{baseUrl}}",
                  ],
                  "path": Array [
                    "crm",
                    "companies",
                  ],
                  "query": Array [],
                  "variable": Array [],
                },
              },
              "response": Array [],
            },
          ],
          "name": "Minimal CRM Variations",
        },
      ],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "type": "any",
      "value": "https://unify.apideck.com",
    },
  ],
}
`;

exports[`TestSuite Variations openApiResponse content-types should use a specific content-type when multiple are available 1`] = `
Object {
  "_": Object {
    "postman_id": "cccccccc-cccc-cccc-cccc-cccccccccccc",
  },
  "event": Array [],
  "info": Object {
    "name": "Minimal CRM",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "List companies",
      "request": Object {
        "header": Array [
          Object {
            "key": "x-apideck-consumer-id",
            "value": "test_consumer",
          },
          Object {
            "key": "x-apideck-app-id",
            "value": "2222",
          },
          Object {
            "key": "x-apideck-service-id",
            "value": "salesforce",
          },
        ],
        "method": "GET",
        "url": Object {
          "host": Array [
            "{{baseUrl}}",
          ],
          "path": Array [
            "crm",
            "companies",
          ],
          "query": Array [],
          "variable": Array [],
        },
      },
      "response": Array [],
    },
    Object {
      "event": Array [],
      "item": Array [
        Object {
          "event": Array [],
          "item": Array [
            Object {
              "event": Array [
                Object {
                  "listen": "test",
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                    "type": "text/javascript",
                  },
                },
              ],
              "name": "List companies[SpecificContentType]",
              "request": Object {
                "header": Array [
                  Object {
                    "key": "x-apideck-consumer-id",
                    "value": "test_consumer",
                  },
                  Object {
                    "key": "x-apideck-app-id",
                    "value": "2222",
                  },
                  Object {
                    "key": "x-apideck-service-id",
                    "value": "salesforce",
                  },
                  Object {
                    "key": "Accept",
                    "value": "application/json",
                  },
                ],
                "method": "GET",
                "name": "List companies[SpecificContentType]",
                "url": Object {
                  "host": Array [
                    "{{baseUrl}}",
                  ],
                  "path": Array [
                    "crm",
                    "companies",
                  ],
                  "query": Array [],
                  "variable": Array [],
                },
              },
              "response": Array [],
            },
          ],
          "name": "Minimal CRM Variations",
        },
      ],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "type": "any",
      "value": "https://unify.apideck.com",
    },
  ],
}
`;

exports[`TestSuite Variations response codes should handle partial wildcard response codes and generate multiple variations 1`] = `
Object {
  "_": Object {
    "postman_id": "cccccccc-cccc-cccc-cccc-cccccccccccc",
  },
  "event": Array [],
  "info": Object {
    "name": "Minimal CRM",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "List companies",
    },
    Object {
      "event": Array [],
      "item": Array [
        Object {
          "event": Array [],
          "item": Array [
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][200][application/xml]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][200][application/json]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][200][text/plain]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 201\\", function () {
    pm.expect(pm.response.code).to.equal(201);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][201][application/xml]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 201\\", function () {
    pm.expect(pm.response.code).to.equal(201);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][201][application/json]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 201\\", function () {
    pm.expect(pm.response.code).to.equal(201);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][201][text/plain]",
            },
          ],
          "name": "Minimal CRM Variations",
        },
      ],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "value": "https://unify.apideck.com",
    },
  ],
}
`;

exports[`TestSuite Variations response codes should handle wildcard response codes and generate multiple variations 1`] = `
Object {
  "_": Object {
    "postman_id": "cccccccc-cccc-cccc-cccc-cccccccccccc",
  },
  "event": Array [],
  "info": Object {
    "name": "Minimal CRM",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "List companies",
    },
    Object {
      "event": Array [],
      "item": Array [
        Object {
          "event": Array [],
          "item": Array [
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][200][application/xml]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][200][application/json]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 200\\", function () {
    pm.expect(pm.response.code).to.equal(200);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][200][text/plain]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 201\\", function () {
    pm.expect(pm.response.code).to.equal(201);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][201][application/xml]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 201\\", function () {
    pm.expect(pm.response.code).to.equal(201);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][201][application/json]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 201\\", function () {
    pm.expect(pm.response.code).to.equal(201);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][201][text/plain]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 400\\", function () {
    pm.expect(pm.response.code).to.equal(400);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][400][application/json]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 400\\", function () {
    pm.expect(pm.response.code).to.equal(400);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][400][application/xml]",
            },
            Object {
              "event": Array [
                Object {
                  "script": Object {
                    "exec": Array [
                      "// Validate response status code 
pm.test(\\"[GET]::/crm/companies - Response status code is 400\\", function () {
    pm.expect(pm.response.code).to.equal(400);
});
",
                    ],
                  },
                },
              ],
              "name": "List companies[WildcardCode][400][text/plain]",
            },
            Object {
              "event": Array [],
              "name": "List companies[WildcardCode][default][application/xml]",
            },
            Object {
              "event": Array [],
              "name": "List companies[WildcardCode][default][application/json]",
            },
            Object {
              "event": Array [],
              "name": "List companies[WildcardCode][default][text/plain]",
            },
          ],
          "name": "Minimal CRM Variations",
        },
      ],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "value": "https://unify.apideck.com",
    },
  ],
}
`;

exports[`TestSuite Variations should generateVariationTests for variations 1`] = `
Object {
  "_": Object {
    "postman_id": "cccccccc-cccc-cccc-cccc-cccccccccccc",
  },
  "event": Array [],
  "info": Object {
    "name": "Minimal CRM",
    "schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json",
    "version": undefined,
  },
  "item": Array [
    Object {
      "event": Array [],
      "name": "List companies",
      "request": Object {
        "header": Array [
          Object {
            "key": "x-apideck-consumer-id",
            "value": "test_consumer",
          },
          Object {
            "key": "x-apideck-app-id",
            "value": "2222",
          },
          Object {
            "key": "x-apideck-service-id",
            "value": "salesforce",
          },
        ],
        "method": "GET",
        "url": Object {
          "host": Array [
            "{{baseUrl}}",
          ],
          "path": Array [
            "crm",
            "companies",
          ],
          "query": Array [],
          "variable": Array [],
        },
      },
      "response": Array [],
    },
    Object {
      "event": Array [],
      "item": Array [],
      "name": "Variation Tests",
    },
  ],
  "variable": Array [
    Object {
      "key": "baseUrl",
      "type": "any",
      "value": "https://unify.apideck.com",
    },
  ],
}
`;
